apps/addons/cron.py

```diff
@@ -454,6 +454,7 @@ def reindex_addons(index=None, addon_type=None):
     ids = (Addon.objects.values_list('id', flat=True)
            .filter(_current_version__isnull=False,
                    status__in=amo.VALID_STATUSES,
+                   is_listed=True,
```

Comment:
Do we also need this in other places like in `addons.tasks.index_addons`?

When we submit a new addon, where/how is it index? I think we need to do something similar there so unlisted addons don't get indexed at all (same for versions and files?).

This could obviously be done in another PR, do you want me to open a new bug?